pmd-core/src/main/java/net/sourceforge/pmd/util/StringUtil.java

```diff
@@ -79,21 +86,50 @@ public static String withoutPrefixes(String text, String... prefixes) {
         return text;
     }
 
+
+    /**
+     * @param value String
+     *
+     * @return boolean
+     */
+    public static boolean isNotEmpty(String value) {
+        return !isEmpty(value);
+    }
+
+
     /**
      * Returns true if the value arg is either null, empty, or full of
      * whitespace characters. More efficient that calling
      * (string).trim().length() == 0
      *
      * @param value
-     * @return <code>true</code> if the value is empty, <code>false</code>
-     *         otherwise.
+     *
+     * @return <code>true</code> if the value is empty, <code>false</code> otherwise.
      */
     public static boolean isEmpty(String value) {
+        return isMissing(value) || nullSafeIsBlank(value);
+    }
+
+
+    /**
+     * Returns true if the argument is null or the empty string.
+     *
+     * @param value String to test
+     *
+     * @return True if the argument is null or the empty string
+     */
+    public static boolean isMissing(String value) {
 
         if (value == null || "".equals(value)) {
             return true;
         }
 
+        return false;
+    }
+
+
+    /** Argument must not be null. */
+    private static boolean nullSafeIsBlank(String value) {
```

Comment:
In my experience, people use "null safe" to indicate the method operates on nulls without issue.  You use it in the reverse here.  `nullUnsafeIsBlank` would be more apropos.